archivers/libarchive/Portfile

```diff
@@ -24,6 +24,13 @@ checksums        rmd160  c59dd17f764e33c2984058091ea00b5768e35ce4 \
 homepage         https://libarchive.org/
 master_sites     ${homepage}downloads/
 
+# This port is a recursive dependency of recent clang versions.
+foreach ver {8.0 7.0 6.0 5.0} {
+    if {![file exists ${prefix}/bin/clang-mp-${ver}]} {
+        compiler.blacklist-append macports-clang-${ver}
+    }
+}
+
```

Comment:
I think its really a bad idea to have to replicate the above block in so many places. What happens when a new clang comes out in the future, 9.0 ? Someone then has to remember to add it to all these lists if versions. What uf the deps change ?

I think the above should be moved to a single place (a PG?) and the ports in question just use that.

Reply:
There are, for instance, a number of PGs that already maintain a list of 'known clang versions'. e.g. `clang_versions` from

https://github.com/macports/macports-ports/blob/master/_resources/port1.0/group/compilers-1.0.tcl

it would need to be made available to portfiles, if not already, but then it could be used instead of the hardcoded lists. 

Better yet, move the entire block to one of these PGs, and add a single option each port then needs to set if it wishes to activate the additional blacklisting.

Reply:
Fair points. I don't think there's a good solution for when a new clang version has new deps; that just has to be part of the process of adding a new clang version (or adding it to the default compiler.fallback at least).

Reply:
yeah, new deps is probably hard to accommodate automatically. Moving the logic to a shared single place though should be easy enough.